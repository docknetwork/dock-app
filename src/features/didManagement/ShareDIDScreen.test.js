import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {ShareDIDScreen} from './ShareDIDScreen';

const mockStore = configureMockStore();

describe('ShareDIDScreen', () => {
  it('should render correctly', () => {
    const initialState = {
      // placeholder for redux store
    };

    const wrapper = shallow(<ShareDIDScreen />, {
      context: {store: mockStore(initialState)},
    });
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
