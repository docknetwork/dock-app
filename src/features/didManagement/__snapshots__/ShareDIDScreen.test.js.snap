// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareDIDScreen should render correctly 1`] = `
<RNCSafeAreaView
  accessibilityLabel="ShareDIDScreen"
  style={
    Object {
      "backgroundColor": "#000",
      "flex": 1,
    }
  }
  testID="ShareDIDScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <BackButton
          accessibilityLabel="BackButton"
          testID="BackButton"
        />
      </Styled(View)>
      <Memo(ForwardRef(Stack))
        alignItems="center"
      >
        <Styled(Component)
          mb={2}
          variant="h1"
        >
          Share DID
        </Styled(Component)>
        <Styled(Component)
          variant="screen-description"
        >
          Share your Decentralized Identifier
        </Styled(Component)>
      </Memo(ForwardRef(Stack))>
      <Styled(ScrollView)
        marginLeft={26}
        marginRight={26}
      >
        <Memo(ForwardRef(Stack))
          alignItems="center"
          bg="#27272A"
          borderRadius={12}
          my={7}
          p={5}
        >
          <Memo(ForwardRef(Stack))
            bg="#fff"
            p={2}
          >
            <QRCode
              size={375}
            />
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(Stack))
            alignItems="center"
            direction="row"
            flex={1}
            pt={5}
          >
            <Memo(ForwardRef(Box))
              pr={2}
            />
            <Memo(ForwardRef(Stack))
              alignItems="flex-start"
              direction="column"
              flex={1}
            >
              <Styled(Component)
                variant="h3"
              />
              <Styled(Component) />
            </Memo(ForwardRef(Stack))>
          </Memo(ForwardRef(Stack))>
        </Memo(ForwardRef(Stack))>
      </Styled(ScrollView)>
      <Memo(ForwardRef(Box))
        mb={7}
        mx={7}
      >
        <Button
          size="sm"
          startIcon={
            <Memo
              as={[Function]}
              name="copy-outline"
              size="sm"
            />
          }
        >
          Copy DID
        </Button>
      </Memo(ForwardRef(Box))>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;
